### PR TITLE
treewide: add meta.mainProgram to many packages

### PR DESCRIPTION
**Pull request by malob**
###### Description of changes

Following on from my efforts in #166865, I refined my hacky script and I've been able to find an additional 167 packages that provide a single binary whose name differs from the package's name in `nixpkgs` (or the package's `pname`) and so don't work with `nix run`.

The `mainProgram` additions in 16e15fa68f0247d1ccc9bf94e2906097bcaa2d5c should be pretty boring and uncontroversial. 

I separated out additions of `mainProgram` to files located in `pkgs/development/libraries/` into a separate commit (9e4be532219685cdd81e4db54f9fc061d250ba25) since a lot of these required more judgement. I think it makes sense for many of these packages to include a `mainProgram` as they often provide a single binary utility as part of the package that folks might want to use. However, in some cases it's pretty marginal. For those cases, I looked into the functionality of the binary and what it's used for, and made a judgment call. (I did err on the side of adding `mainProgram` in a lot of the marginal cases since it does little to no harm, but there were many cases where I left it off since it made no sense.)

Finally I made separate commit for `nimble-unwrapped` (d8da6a153b7951f55f4e0243047ad4ae644b4ed5) since I added a new `meta` attribute set since it was missing entirely.

I've gone through these changes pretty carefully, but a couple more skims for others is probably in order as a sanity check.

(While I was at it, I also ended up tweaking the order of the `meta` attributes to my understanding of which order is best practice, as well as updating some of the homepages and adding some when they were missing.)

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by siraben** [comment]
Result of `nixpkgs-review pr 171602` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)

File: pkgs/tools/admin/docker-credential-helpers/default.nix [hunk 1]
```diff
@@ -45,5 +45,7 @@ buildGoPackage rec {
     license = licenses.mit;
     maintainers = [ maintainers.marsam ];
     platforms = platforms.linux ++ platforms.darwin;
+  } // lib.optionalAttrs stdenv.isDarwin {
+    mainProgram = "docker-credential-osxkeychain";
```
Comment (lovesegfault) [inline pkgs/tools/admin/docker-credential-helpers/default.nix:49]: Could you move this to a separate PR? I think having platform-dependent `mainProgram`s warrants some debate.

Comment (malob) [reply pkgs/tools/admin/docker-credential-helpers/default.nix]: Will do.

Comment (malob) [reply pkgs/tools/admin/docker-credential-helpers/default.nix]: #171749

**Comment by malob** [comment]
Result of `nixpkgs-review pr 171602` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
